xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-default/src/main/java/org/xwiki/model/validation/internal/EntityNameValidationConfigurationListener.java

```diff
@@ -0,0 +1,119 @@
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.model.validation.internal;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.bridge.event.DocumentCreatedEvent;
+import org.xwiki.bridge.event.DocumentDeletedEvent;
+import org.xwiki.bridge.event.DocumentUpdatedEvent;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.model.reference.LocalDocumentReference;
+import org.xwiki.model.validation.EntityNameValidation;
+import org.xwiki.model.validation.EntityNameValidationManager;
+import org.xwiki.observation.EventListener;
+import org.xwiki.observation.event.Event;
+import org.xwiki.observation.event.filter.RegexEventFilter;
+
+import com.xpn.xwiki.doc.XWikiDocument;
+
+/**
+ * Listen for create/update/delete events on the entity name validation classes and objects. When their values change,
+ * clears the entity name validators caches (by calling {@link EntityNameValidation#cleanConfigurationCache(String)} on
+ * the registered {@link EntityNameValidation} components). The cache is managed per wiki, and only the cache of the
+ * current wiki is cleared.
+ *
+ * @version $Id$
+ * @since 13.4
+ * @since 12.10.8
+ */
+@Component
+@Named(EntityNameValidationConfigurationListener.NAME)
+@Singleton
+public class EntityNameValidationConfigurationListener implements EventListener
+{
+    /**
+     * The name of the event listener.
+     */
+    static final String NAME = "EntityNameValidationConfigurationListener";
+
+    @Inject
+    private EntityNameValidationManager entityNameValidationManager;
+
+    @Inject
+    @Named("local")
+    private EntityReferenceSerializer<String> entityReferenceSerializer;
+
+    @Override
+    public String getName()
+    {
+        return NAME;
+    }
+
+    @Override
+    public List<Event> getEvents()
+    {
+        LocalDocumentReference classReference = EntityNameValidationConfigurationSource.CLASS_REFERENCE;
+        LocalDocumentReference documentReference = EntityNameValidationConfigurationSource.DOC_REFERENCE;
+        RegexEventFilter configurationClassEventFilter = createRegexEventFilter(classReference);
+        RegexEventFilter configurationObjectEventFilter = createRegexEventFilter(documentReference);
+        return Arrays.asList(
```

Comment:
I don't understand why you're listening on those events, IMO you could just listen on the `XObjectAdded/Updated/DeletedEvent` with a `XWiki.EntityNameValidation.ConfigurationClass` xobject reference.

Reply:
> The rational is to listen to the single class and object of this configuration (does not make sens to invalidate on an XObject that will never be read), on all the wikis. 

Actually what I don't get is why you're listening on the configuration class document? Even if this one changes it doesn't make the config itself changing.

Reply:
In a minimal configuration, the XObject can be missing, or one of the properties left blank. In this case is it interesting to rely on the XClass defaults values.
This is the can on docker tests with a minimal distribution for instance.

Reply:
bq. In this case is it interesting to rely on the XClass defaults values.

well maybe I missed the change but the default values are empty in the xclass, that's the configuration components implementation which gives default values.

Reply:
My mistake, I don't know why I though XClass properties could define default values... Removing.

Reply:
> Could be interesting to add two keys in xwiki.properties to provide default values for the configuration, That would also make testing easier.

That's actually an option that I'm thinking about to actually fix the problem I raised in my comment on https://jira.xwiki.org/browse/XWIKI-18670?focusedCommentId=109158&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-109158